src/java.base/share/classes/jdk/internal/access/JavaLangAccess.java

```diff
@@ -296,46 +296,50 @@ public interface JavaLangAccess {
 
     /**
      * Count the number of leading positive bytes in the range.
+     * <p>
+     * <b>WARNING: This method does not perform any bound checks.</b>
      */
-    int countPositives(byte[] ba, int off, int len);
+    int uncheckedCountPositives(byte[] ba, int off, int len);
 
     /**
      * Count the number of leading non-zero ascii chars in the String.
      */
     int countNonZeroAscii(String s);
 
     /**
-     * Constructs a new {@code String} by decoding the specified subarray of
-     * bytes using the specified {@linkplain java.nio.charset.Charset charset}.
-     *
-     * The caller of this method shall relinquish and transfer the ownership of
-     * the byte array to the callee since the later will not make a copy.
+     * Constructs a new {@code String} by decoding the specified byte array
+     * using the specified {@linkplain java.nio.charset.Charset charset}.
+     * <p>
+     * <b>WARNING: The caller of this method shall relinquish and transfer the
+     * ownership of the byte array to the callee</b>, since the latter will not
+     * make a copy.
      *
      * @param bytes the byte array source
      * @param cs the Charset
      * @return the newly created string
      * @throws CharacterCodingException for malformed or unmappable bytes
      */
-    String newStringNoRepl(byte[] bytes, Charset cs) throws CharacterCodingException;
+    String uncheckedNewStringNoRepl(byte[] bytes, Charset cs) throws CharacterCodingException;
 
     /**
-     * Encode the given string into a sequence of bytes using the specified Charset.
-     *
-     * This method avoids copying the String's internal representation if the input
-     * is ASCII.
-     *
-     * This method throws CharacterCodingException instead of replacing when
-     * malformed input or unmappable characters are encountered.
+     * Encode the given string into a sequence of bytes using the specified
+     * {@linkplain java.nio.charset.Charset charset}.
+     * <p>
+     * <b>WARNING: This method return the {@code byte[]} backing the provided
```

Comment:
```suggestion
     * <b>WARNING: This method returns the {@code byte[]} backing the provided
```
Should this more explicitly say the returned byte array must not be modified?

Reply:
Improved as suggested in cd156fbab46.